### PR TITLE
Add GitHub connector

### LINKED ISSUE #229: GitHub Connector
As the Parsons repository sees more usage and contributions, it would be great to be able to better report on things like outside contributions and automatically track progress on Parsons issues. To start, we would love to be able to pull down data about Pull Requests and Issues.

### PR DESCRIPTION
**Pull request by pjsier**
Should close #229. Adds an initial GitHub connector that's a thin wrapper around [PyGithub](https://github.com/PyGithub/PyGithub).

There's a ton of functionality, so this is an initial stab at a setup that we could build on more. A few things I wanted to flag:

* Right now it loads pull requests in any call to `list_repo_issues`, and I'm not seeing a clear way of filtering that out through the API. If it's a priority we might be able to do it manually
* A lot of the JSON responses are nested, and I'm not sure how the `Table` will handle that

Let me know what updates I can make, thanks!

### CODE REVIEW

File: parsons/github/github.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+        Returns:
+            ``Table``
+                Table object created from the raw data of the list
+        """
+
+        page_start = (page - 1) * page_size
+        page_end = page_start + page_size
+
+        return Table([
```
[the remark this answers is not in the document]
Comment (pjsier) [reply parsons/github/github.py:66]: That works! It's actually the default for PyGithub, so it should be straightforward to implement here

File: parsons/github/github.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        self.username = check_env.check('GITHUB_USERNAME', username, optional=True)
+        self.password = check_env.check('GITHUB_PASSWORD', password, optional=True)
+        self.access_token = check_env.check('GITHUB_ACCESS_TOKEN', access_token, optional=True)
+
+        if self.username and self.password:
+            self.client = PyGithub(self.username, self.password)
+        elif self.access_token:
+            self.client = PyGithub(self.access_token)
+        else:
+            raise KeyError(
```
[the remark this answers is not in the document]
Comment (pjsier) [reply parsons/github/github.py:41]: Good point! I didn't think about that, but that works for me.

For the 404 error I should be able to use a decorator that wraps the entire class based off of what I was working on before unless you can think of a more straightforward way

**Comment by pjsier** [comment]
@eliotst made some updates based on your review, let me know what you think!

File: parsons/github/github.py [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+                Full repo name (account/name)
+
+        Returns:
+            dict
+                Repo information
+        """
+
+        return self.client.get_repo(repo_name).raw_data
+
+    def list_user_repos(self, username, page=1, page_size=25):
```
[the remark this answers is not in the document]
Comment (eliotst) [reply parsons/github/github.py]: I realize this comment is less pressing in a world where you default to pull everything down.

**Pull request merged**